spec/scripts/changelog_check_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+require_relative '../../scripts/changelog_check'
+
+GIT_FIXTURES = YAML.safe_load(File.read(File.expand_path('spec/fixtures/git_log_changelog.yml')))
+
+describe '#build_changelog' do
+  it 'builds a git log into structured changelog objects' do
+    git_log = GIT_FIXTURES.values.map { |x| x['commit_log'] }.join("\n")
+    changelog_entries = generate_changelog(git_log)
+    expect(changelog_entries.length).to eq 2
+    fixture_and_changelog = GIT_FIXTURES.values.filter do |x|
+      x['category'].present?
+    end.zip(changelog_entries)
+
+    fixture_and_changelog.each do |fixture, changelog|
+      expect(fixture['category']).to eq changelog.category
+      expect(fixture['subcategory']).to eq changelog.subcategory
+      expect(fixture['pr_number']).to eq changelog.pr_number
+      expect(fixture['change']).to eq changelog.change
+    end
+  end
+
+  it 'skips commits with [skip changelog]' do
+    commits = [
+      GIT_FIXTURES['squashed_commit_with_one_commit'],
+      GIT_FIXTURES['squashed_commit_with_skip'],
+    ]
+    git_log = commits.map { |x| x['commit_log'] }.join("\n")
+    changelog = generate_changelog(git_log)
+    expect(changelog.length).to eq 1
+
+    expect(commits.first['category']).to eq changelog.first.category
+    expect(commits.first['subcategory']).to eq changelog.first.subcategory
+    expect(commits.first['pr_number']).to eq changelog.first.pr_number
+    expect(commits.first['change']).to eq changelog.first.change
+  end
+end
+
+describe '#build_structured_git_log' do
+  it 'builds a git log into structured objects' do
+    GIT_FIXTURES.values.each do |commit_fixture|
+      commits = build_structured_git_log(commit_fixture['commit_log'])
+      expect(commits.length).to eq 1
+      expect(commits.first.title).to eq commit_fixture['title']
+      expect(commits.first.commit_messages).to eq commit_fixture['commit_messages']
+    end
+  end
+end
+
+describe '#generate_invalid_changes' do
+  it 'returns changelog entries without a valid structure' do
+    commits = [
+      GIT_FIXTURES['squashed_commit_invalid'],
+      GIT_FIXTURES['squashed_commit_with_one_commit'],
+      GIT_FIXTURES['squashed_commit_with_skip'],
+    ]
+    git_log = commits.map { |x| x['commit_log'] }.join("\n")
```

Comment:
I think rubocop might yell at this and suggest using `pluck` if ActiveSupport is loaded
```suggestion
    git_log = commits.pluck('commit_log').join("\n")
```

(personally I don't think it's that clear, maybe we even disable it...)

Reply:
Hm, we do have it [enabled](https://github.com/18F/identity-idp/blob/3f952455363c5d6851a9eea65a9f0d97c31d97e3/.rubocop.yml#L819-L820), but rubocop didn't say anything when I ran it locally.

`git_log = commits.map { |x| x[:commit_log] }.join("\n")` is caught, so it must only work for atom keys?

Reply:
I did switch all of them to `pluck` though